cmd/adapter/adapter.go

```diff
@@ -83,6 +85,8 @@ func (cmd *PrometheusAdapter) addFlags() {
 			"and custom metrics API resources")
 	cmd.Flags().DurationVar(&cmd.MetricsRelistInterval, "metrics-relist-interval", cmd.MetricsRelistInterval, ""+
 		"interval at which to re-list the set of all available metrics from Prometheus")
+	cmd.Flags().DurationVar(&cmd.MetricsStartDuration, "metrics-start-duration", cmd.MetricsStartDuration, ""+
```

Comment:
can we default this to something sane based the `metrics-relist-interval` so that we don't confuse people  who are expecting the existing behavior?  e.g. if someone sets `metrics-relist-interval` to 20 minutes, they probably don't expect the max metrics age to be 10 minutes, and if someone sets it to 30 seconds, they probably also don't expect the age to be 10 minutes.  Maybe `2 * relist interval` by default or something?

Reply:
Agreed that I can make it longer than `metrics-relist-interval` by default, i.e. `20m`. That would suggest that max age should be at least twice as high as relist interval.

Why setting max age to `10m` with relist interval `30s` is bad though? I think it's perfectly fine actually :) e.g. even if Prometheus is configured to be super slow and refreshes metrics once every `10m`, I still don't want to have a potential delay of `9m:59s` once a new metric is discovered by Prometheus, so I will set it to `30s`.

Reply:
@DirectXMan12 I can probably add a validation code that would require the `max-age >= 2 * relist-interval`?

Reply:
no validation needed.  My main concern was that it means metrics that are no longer valid will show up in discovery for a lot longer.  If you have things coming and going quickly, this could lead to a lot more metrics to process during discovery than you intend.  It's not a big deal here, though, and we can always follow up later.